nestris_ocr/capturing/stream.py

```diff
@@ -0,0 +1,71 @@
+import cv2
+from collections import deque
+from multiprocessing import Lock
+from multiprocessing.pool import ThreadPool
+
+from PIL import Image
+from typing import Tuple
+
+from nestris_ocr.capturing.base import AbstractCapture
+from nestris_ocr.utils import xywh_to_ltrb
+from nestris_ocr.types import XYWHBox
+
+FRAME_BUFFER = 150
+
+
+class StreamCapture(AbstractCapture):
+    def __init__(self, source_id: str, xywh_box: XYWHBox, extra_data: str) -> None:
+        super().__init__(source_id, xywh_box, extra_data)
+
+        print("Initializing stream URL", source_id)
+
+        self.cap = cv2.VideoCapture(source_id)
+        self.frame_buffer = deque([], FRAME_BUFFER)
+        self.cv2_retval = None
+
+        self.running = False
+        self.read_lock = Lock()
+        self.start()
+
+    def get_image(self, rgb: bool = False) -> Tuple[float, Image.Image]:
+        if not self.cv2_retval:
+            raise Exception("Faulty capturing device")
+
+        with self.read_lock:
+            cv2_image = self.frame_buffer.popleft()
+
+        if rgb:
+            # can we do this in the thread pool?
```

Comment:
Answered in discord; but since we are using thread (not process) it doesnt matter too much where we RGB it. Also if the stream is h264, it might *already* be rgb output (though its funneling through opencv, so its probably BGR). You could try displaying the image with rgb=false, and if its rgb you can remove these lines, if its bgr keep them. Not a big deal for this PR at all, i wouldn't bother testing it till later.